### PR TITLE
use default handler

### PR DESCRIPTION
**Pull request by deads2k**
Builds on already lgtm'd pulls.  Only the last commit is unique.

The biggest impact is to audit.

@soltysh @sttts you know want it before dcut today. :)

### CODE REVIEW

File: pkg/cmd/server/origin/master.go [hunk 1]
```diff
@@ -285,45 +276,10 @@ func (c *MasterConfig) buildHandlerChain() (func(apiHandler http.Handler, kc *ap
...
-			handler = apiserverfilters.WithTimeoutForNonLongRunningRequests(handler, genericConfig.RequestContextMapper, genericConfig.LongRunningFunc)
-			// TODO: MaxRequestsInFlight should be subdivided by intent, type of behavior, and speed of
-			// execution - updates vs reads, long reads vs short reads, fat reads vs skinny reads.
-			// NOTE: read vs. write is implemented in Kube 1.6+
-			handler = apiserverfilters.WithMaxInFlightLimit(handler, genericConfig.MaxRequestsInFlight, genericConfig.MaxMutatingRequestsInFlight, genericConfig.RequestContextMapper, genericConfig.LongRunningFunc)
-			handler = apifilters.WithRequestInfo(handler, apiserver.NewRequestInfoResolver(genericConfig), genericConfig.RequestContextMapper)
-			handler = apirequest.WithRequestContext(handler, genericConfig.RequestContextMapper)
-			handler = apiserverfilters.WithPanicRecovery(handler)
-			///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+			handler = apiserver.DefaultBuildHandlerChain(handler, genericConfig)
```
Comment (deads2k) [inline pkg/cmd/server/origin/master.go]: @sttts as a past rebaser, this should make you very happy.

File: pkg/cmd/server/kubernetes/master/master_config.go [hunk 2]
```diff
@@ -489,6 +510,29 @@ func buildKubeApiserverConfig(
 		return nil, err
 	}
 
+	// we don't use this anymore
+	genericConfig.LegacyAuditWriter = nil
+	if masterConfig.AuditConfig.Enabled {
```
Comment (deads2k) [inline pkg/cmd/server/kubernetes/master/master_config.go]: @soltysh review carefully

File: pkg/cmd/server/kubernetes/master/master_config.go [hunk 3]
```diff
@@ -159,6 +167,18 @@ func BuildKubeAPIserverOptions(masterConfig configapi.MasterConfig) (*kapiserver
 	server.KubeletConfig.ReadOnlyPort = 0
 
 	// resolve extended arguments
+	args := map[string][]string{}
+	for k, v := range masterConfig.KubernetesMasterConfig.APIServerArguments {
+		args[k] = v
+	}
+	if masterConfig.AuditConfig.Enabled {
```
Comment (deads2k) [inline pkg/cmd/server/kubernetes/master/master_config.go]: @soltysh review carefully

File: pkg/cmd/server/origin/master.go (hunk 1 above)
Comment (sttts) [reply pkg/cmd/server/origin/master.go]: ❤️

File: pkg/cmd/server/kubernetes/master/master_config.go (hunk 3 above)
Comment (soltysh) [reply pkg/cmd/server/kubernetes/master/master_config.go]: 👍

File: pkg/cmd/server/kubernetes/master/master_config.go [hunk 4]
```diff
@@ -492,6 +510,29 @@ func buildKubeApiserverConfig(
 		return nil, err
 	}
 
+	// we don't use this anymore
```
Comment (soltysh) [inline pkg/cmd/server/kubernetes/master/master_config.go]: I'll probably update this comment to explicitly mention we don't use legacy audit at all. All auditing in origin uses the new advanced feature.

File: pkg/cmd/server/kubernetes/master/master_config.go (hunk 2 above)
Comment (soltysh) [reply pkg/cmd/server/kubernetes/master/master_config.go]: 👍

**Comment by soltysh** [comment]
Rebase and ship it!
/lgtm

File: pkg/cmd/server/kubernetes/master/master_config.go (hunk 4 above)
Comment (deads2k) [reply pkg/cmd/server/kubernetes/master/master_config.go]: done

**Pull request merged**